utils/common.go

```diff
@@ -0,0 +1,83 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"os"
+
+	"github.com/blndgs/model"
+	"github.com/spf13/cobra"
+)
+
+// AddCommonFlags adds common flags to the provided Cobra command.
+// It adds a string flag 'userop' for user operation JSON and
+// a boolean flag 'zerogas' to enable zero gas mode.
+func AddCommonFlags(cmd *cobra.Command) {
+	cmd.Flags().String("userop", "", "User operation JSON")
+	cmd.Flags().Bool("zerogas", false, "Use zero gas mode")
+
+	// Mark the 'userop' flag as required
+	if err := cmd.MarkFlagRequired("userop"); err != nil {
+		panic(err)
+	}
+}
+
+// IsZeroGas checks if the 'zerogas' flag is set in the command.
+// It returns true if the 'zerogas' flag is set, otherwise false.
+func IsZeroGas(cmd *cobra.Command) bool {
+	zeroGas, _ := cmd.Flags().GetBool("zerogas")
+	return zeroGas
+}
+
+// GetUserOps parses the 'userop' JSON string or file provided in the command flags
+// and returns a UserOperation object. It panics if the JSON string is empty,
+// the file can't be read, or the JSON can't be parsed.
+func GetUserOps(cmd *cobra.Command) *model.UserOperation {
+	userOpInput, _ := cmd.Flags().GetString("userop")
+
+	var userOpJSON string
+	if fileExists(userOpInput) {
+		// Read JSON from file
+		fileContent, err := os.ReadFile(userOpInput)
+		if err != nil {
+			panic(fmt.Errorf("error reading user operation file: %v", err))
+		}
+		userOpJSON = string(fileContent)
+	} else {
+		// Use input as JSON string
+		userOpJSON = userOpInput
+	}
+
+	if userOpJSON == "" {
+		panic("user operation JSON is required")
+	}
+
+	var userOp model.UserOperation
+	err := json.Unmarshal([]byte(userOpJSON), &userOp)
+	if err != nil {
+		panic(fmt.Errorf("error parsing user operation JSON: %v", err))
+	}
+	return &userOp
+}
+
+// UpdateUserOp updates the given user operation based on the provided nonce and zeroGas flag.
+// If zeroGas is true, all gas-related fields in the user operation are set to zero.
+// The function returns the updated UserOperation object.
+func UpdateUserOp(userOp *model.UserOperation, nonce *big.Int, zeroGas bool) *model.UserOperation {
+	if zeroGas {
```

Comment:
Setting `MaxFeePerGas` and `MaxPriorityFeePerGas` to 0 results in an effective zero gas setting. For the remaining 3 limits in the zero setting, hardcode please to the the max limit of 300k.